### ISSUE TITLE
Help wanted : Put Checkbox in headers ...

### ISSUE THREAD

**Issue by fabricecouvert** [olifolkerd/tabulator#50]
Hi ,
I'm starting using tabulator as it's a really cool plugin very easy and 'tunable' but i m stuck on this problem.....
Is there any possibilty to add a checkbox in the header of a column to implement a 'select all' behavior'??
thanks in advance....

**Comment by olifolkerd** [comment]
At the moment there is no built-in row selection functions, though this will be comming in the next couple of months.

In the mean time, the best bet would be to use a custom formatter to put a checkbox input element in the first cell in each row then use that to toggle a ***"selected"*** class on the row that you can then use to style and access selected tows. 

To insert an input in header you can pass html into the title filed.

Below is an rough example of how to achieve it:

```js
var selectionFormatter = function(value, data, cell, row, options, formatterParams){
     var rowSelect = $("<input type='checkbox' class='row-select>");
    
    rowSelect.on("change", function(){
        if($(this).is(":checked")){
            $(this).closest(".tabulator-row").addClass("selected");
        }else{
            $(this).closest(".tabulator-row").removeClass("selected");
        }
    })
 
     return rowSelect;
}

$("#example-table").tabulator({
    columns:[
        {title:"<input type='checkbox' class='select-all'>", formatter:selectionFormatter}, //row selector
        {title:"Name", field:"name"}, //data comlumn
    ]
});

$("#example-table .select-all").on("change", function(){
    if($(this).is(":checked")){
        $("#example-table .row-select").prop("checked", true).closest(".tabulator-row").addClass("selected");
    }else{
        $("#example-table .row-select").prop("checked", false).closest(".tabulator-row").removeClass("selected");
    }
});

```

**Comment by fabricecouvert** [comment]
thanks for that quick response.
you are really doing a great job.
one other thing that may be also cool is to add an example on how catch events raised by tabultator instance in documentation because there is an exhaustive listof all of them but at the end of the doc but one example of implementation may help...(in my case to maintain checked state of rows when filtering data...)

**Comment by olifolkerd** [comment]
Thanks, it is always great to hear people are enjoying using tabulator.

There is an example of some of the callbacks being used in the example page [Callback Section](http://olifolkerd.github.io/tabulator/examples/#callbacks)

If you want to maintain the checked state of the rows, wou will need to store the selected property in the row data.I would suggest adding a ***field*** value of "checked" to the selector column definition  and then adding a checkbox editor to handle the the change of ticked to unticked, that way the selection value will be stored with the row data.

if i et a chance this afternoon i will put togeater a bit of sample code for you.

**Comment by fabricecouvert** [comment]
Thanks, for your response...
In fact I know how to use those  handlers..
My question was more about how to catch an event like ''renderComplete"....

**Comment by fabricecouvert** [comment]
Hi,
in the doc the example for the dataFiltered callback is using the dataLoaded property...is that an error ?

Data Filtered
The dataFiltered callback is triggered whenever the table dataset is filtered. It can be set globally by setting the dataFiltered option when you create your Tabulator.

$("#example-table").tabulator({
    dataLoaded:function(data){
        //data - the subset of the total table data that has passed the filter and is now visible
    },
});

**Comment by olifolkerd** [comment]
Well spotted, that is indeed an error! i have just uploaded an update to the documentation.

I can see now how the documentation is confusing when it comes to events. You can add a callback function to an event in the same way as any other callback, the difference being that events do not pass and additional arguments to the callback.

The events are primarily there to help notify the user at every stage of processing.

I will update the documentation this weekend to make events part of the callback section and clarify how to call them.

Thanks for bringing the issue to my attention.

**Comment by olifolkerd** [comment]
I have just released version 2.5 with a completely overhauled event system.

There are loads of updates to old callbacks and a whole host of new callbacks to make interacting with Tabulator even easier.

More information on the changes can be found [here](http://olifolkerd.github.io/tabulator/docs/#callbacks).

**Comment by fabricecouvert** [comment]
Cool :)
I'll take a look as soon as possible.
Thanks for the update .

2016-09-10 14:40 GMT+02:00 Oli Folkerd <notifications@github.com>:
